crates/re_sdk/src/spawn.rs

```diff
@@ -0,0 +1,166 @@
+/// ```
+#[derive(Debug, Clone, Default)]
+pub struct SpawnOptions {
+    /// The port to listen on.
+    ///
+    /// Defaults to `9876` if unspecified.
+    pub port: Option<u16>,
+
+    /// An upper limit on how much memory the Rerun Viewer should use.
+    /// When this limit is reached, Rerun will drop the oldest data.
+    /// Example: `16GB` or `50%` (of system total).
+    ///
+    /// Defaults to `75%` if unspecified.
+    pub memory_limit: Option<String>,
+
+    /// Specifies the name of the Rerun executable.
+    ///
+    /// You can omit the `.exe` suffix on Windows.
+    ///
+    /// Defaults to `rerun` if unspecified.
+    pub executable_name: Option<String>,
+
+    /// Enforce a specific executable to use instead of searching though PATH
+    /// for [`Self::executable_name`].
+    ///
+    /// Unspecified by default.
+    pub executable_path: Option<String>,
+}
+
+impl SpawnOptions {
+    /// Refer to field-level documentation.
+    pub fn with_port(mut self, port: impl Into<u16>) -> Self {
+        self.port = Some(port.into());
+        self
+    }
+
+    /// Refer to field-level documentation.
+    pub fn with_memory_limit(mut self, memory_limit: impl AsRef<str>) -> Self {
+        self.memory_limit = Some(memory_limit.as_ref().to_owned());
+        self
+    }
+
+    /// Refer to field-level documentation.
+    pub fn with_executable_name(mut self, executable_name: impl AsRef<str>) -> Self {
+        self.executable_name = Some(executable_name.as_ref().to_owned());
+        self
+    }
+
+    /// Refer to field-level documentation.
+    pub fn with_executable_path(mut self, executable_path: impl AsRef<str>) -> Self {
+        self.executable_path = Some(executable_path.as_ref().to_owned());
+        self
+    }
+}
+
+impl SpawnOptions {
+    /// Resolves the final port value.
+    pub fn port(&self) -> u16 {
+        self.port.unwrap_or(9876)
+    }
+
+    /// Resolves the final connect address value.
+    pub fn connect_addr(&self) -> std::net::SocketAddr {
+        std::net::SocketAddr::new("127.0.0.1".parse().unwrap(), self.port())
+    }
+
+    /// Resolves the final listen address value.
+    pub fn listen_addr(&self) -> std::net::SocketAddr {
+        std::net::SocketAddr::new("0.0.0.0".parse().unwrap(), self.port())
+    }
+
+    /// Resolves the final memory limit value.
+    pub fn memory_limit(&self) -> String {
+        self.memory_limit.as_deref().unwrap_or("75%").to_owned()
+    }
+
+    /// Resolves the final executable path.
+    pub fn executable_path(&self) -> String {
+        // NOTE: No need for .exe extension on windows.
+        const RERUN_BINARY: &str = "rerun";
+
+        if let Some(path) = self.executable_path.as_deref() {
+            return path.to_owned();
+        }
+
+        self.executable_name
+            .as_deref()
+            .unwrap_or(RERUN_BINARY)
+            .to_owned()
+    }
+}
+
+/// Spawns a new Rerun Viewer process ready to listen for TCP connections.
```

Comment:
This misses the fact that it will first try to connect to an existing viewer, if there is one